File: NoteBook/NoteBook/WriteNoteBookToFile.cs
using System;

namespace NoteBook
{
  /// <summary>
  /// This class write notebook to file
  /// </summary>
  public class WriteNoteBookToFile: Command
  {
    String path;
    NoteBookProvider noteBookProvider;

    /// <summary>
    /// this constructor assigns fields
    /// </summary>
    /// <param name="provider">value interested note book provider</param>
    /// <param name="path">path of file in which we write notrbook</param>
    public WriteNoteBookToFile(NoteBookProvider provider, String path)
    {
      noteBookProvider = provider;
      this.path = path;
    }

    /// <summary>
    /// This method of command
    /// </summary>
    public void Execute()
    {
      noteBookProvider.WriteNoteBookToFile(path);
    }
  }
}

File: TaskNoteBook/src/main/java/WriteNoteBookToFile.java
/**
 * Writes the NoteBook to the file.
 */
public class WriteNoteBookToFile implements ICommand {
  private NoteBookProvider receiver;

  /**
   * Constructor for this class.
   *
   * @param receiver used by command pattern
   */
  public WriteNoteBookToFile(NoteBookProvider receiver) {
    this.receiver = receiver;
  }

  /**
   * Writes the NoteBook to the file.
   */
  @Override
  public void execute() {
    receiver.writeNoteBookToFile();
  }
}
